poller-service.py

```diff
@@ -0,0 +1,286 @@
+    db_server = config['db_host']
+    db_port = 0
+elif ':' in config['db_host']:
+    db_server = config['db_host'].rsplit(':')[0]
+    db_port = int(config['db_host'].rsplit(':')[1])
+else:
+    db_server = config['db_host']
+    db_port = 0
+
+db_dbname = config['db_name']
+
+
+try:
+    amount_of_workers = int(config['poller_service_workers'])
+    if amount_of_workers == 0:
+        amount_of_workers = 16
+except KeyError:
+    amount_of_workers = 16
+
+try:
+    poll_frequency = int(config['poller_service_poll_frequency'])
+    if poll_frequency == 0:
+        poll_frequency = 300
+except KeyError:
+    poll_frequency = 300
+
+try:
+    discover_frequency = int(config['poller_service_discover_frequency'])
+    if discover_frequency == 0:
+        discover_frequency = 21600
+except KeyError:
+    discover_frequency = 21600
+
+try:
+    down_retry = int(config['poller_service_down_retry'])
+    if down_retry == 0:
+        down_retry = 60
+except KeyError:
+    down_retry = 60
+
+try:
+    if db_port == 0:
+        db = MySQLdb.connect(host=db_server, user=db_username, passwd=db_password, db=db_dbname)
+    else:
+        db = MySQLdb.connect(host=db_server, port=db_port, user=db_username, passwd=db_password, db=db_dbname)
+    db.autocommit(True)
+    cursor = db.cursor()
+except:
+    log.critical("ERROR: Could not connect to MySQL database!")
+    sys.exit(2)
+
+
+def poll_worker(device_id, action):
+    try:
+        start_time = time.time()
+        path = poller_path
+        if action == 'discovery':
+            path = discover_path
+        command = "/usr/bin/env php %s -h %s >> /dev/null 2>&1" % (path, device_id)
+        subprocess.check_call(command, shell=True)
+        elapsed_time = int(time.time() - start_time)
+        if elapsed_time < 300:
+            log.debug("DEBUG: worker finished %s of device %s in %s seconds" % (action, device_id, elapsed_time))
+        else:
+            log.warning("WARNING: worker finished %s of device %s in %s seconds" % (action, device_id, elapsed_time))
+    except (KeyboardInterrupt, SystemExit):
+        raise
+    except:
+        pass
+
+
+def lockFree(lock):
+    global cursor
+    query = "SELECT IS_FREE_LOCK('{}')".format(lock)
+    cursor.execute(query)
+    return cursor.fetchall()[0][0] == 1
+
+
+def getLock(lock):
+    global cursor
+    query = "SELECT GET_LOCK('{}', 0)".format(lock)
+    cursor.execute(query)
+    return cursor.fetchall()[0][0] == 1
+
+
+def releaseLock(lock):
+    global cursor
+    query = "SELECT RELEASE_LOCK('{}')".format(lock)
+    cursor.execute(query)
+    return cursor.fetchall()[0][0] == 1
+
+
+def sleep_until(timestamp):
+    now = datetime.now()
+    if timestamp > now:
+        sleeptime = (timestamp - now).seconds
+    else:
+        sleeptime = 0
+    time.sleep(sleeptime)
+
+poller_group = ('and poller_group IN({}) '
+                .format(str(config['distributed_poller_group'])) if 'distributed_poller_group' in config else '')
```

Comment:
Else the string poller_group is an empty string. A less terse way to write this would be:

```python
if 'distributed_poller_group' in config:
    poller_group = 'and poller_group IN(' + str(config['distributed_poller_group']) + ') '
else:
    poller_group = ''
```